python/lsst/ip/isr/isrTask.py

```diff
@@ -928,9 +939,25 @@ def overscanCorrection(self, exposure, amp):
             self.log.info("No Overscan region. Not performing Overscan Correction.")
             return None
 
-        maskedImage = exposure.getMaskedImage()
-        dataView = maskedImage.Factory(maskedImage, amp.getRawDataBBox())
-        overscanImage = maskedImage.Factory(maskedImage, amp.getRawHorizontalOverscanBBox())
+        oscanBBox = amp.getRawHorizontalOverscanBBox()
+
+        # afw.cameraGeom.assembleImage.makeUpdatedDetector doesn't update readoutCorner; DM-15559
+        x0, x1 = oscanBBox.getBeginX(), oscanBBox.getEndX()
+
+        prescanBBox = amp.getRawPrescanBBox()
+        if oscanBBox.getBeginX() > prescanBBox.getBeginX():  # amp is at the right
+            x0 += self.config.overscanNumLeadingColumnsToSkip
+            x1 -= self.config.overscanNumTrailingColumnsToSkip
+        else:
+            x0 += self.config.overscanNumTrailingColumnsToSkip
+            x1 -= self.config.overscanNumLeadingColumnsToSkip
+
+        oscanBBox = afwGeom.BoxI(afwGeom.PointI(x0, oscanBBox.getBeginY()),
+                                 afwGeom.PointI(x1 - 1, oscanBBox.getEndY() - 1))
```

Comment:
I haven't played with this stuff to know if the `-1`s here are right. As long as you're sure that's correct then I think this looks good.

Reply:
This is correct; integer BBoxes are inclusive (otherwise integer and float ones are too different).  It's a bit confusing, but Jim and I convinced ourselves that it was the least confusing option